client/src/redux/slices/schedule.js

```diff
@@ -0,0 +1,67 @@
+import { createSlice } from '@reduxjs/toolkit';
+import request from '../request'; // function that sends requests to server, takes object
+import { wait } from '../../utils'; // simulates server lag response time
+
+// set initial state
+const initialState = {
+    games: [],
+    isLoading: true,
+};
+
+// set/export reducers in object with name and initial state
+export const scheduleSlice = createSlice({
+    name: 'schedule',
+    initialState,
+    reducers: {
+        getInit: state => {
+            console.log('IS LOADING:', state.isLoading);
+            state.isLoading = true;
+        },
+        // reducer takes in state and payload
+        // updates state with payload
+        getScheduleSuccess: (state, { payload }) => {
+            console.log('SCHEDULE STATE', state);
+            console.log('SCHEDULE PAYLOAD', payload);
+            // keep old schedule, add games from schedule
+            // state.games = [...state.games, ...payload];
+            state.games = [...payload];
+            console.log('STATE.GAMES', state.games);
+            state.isLoading = false;
+        },
+    },
+});
+
+// export reducers in .actions
+export const { getInit, getScheduleSuccess } = scheduleSlice.actions;
+
+// function currying?
+export const getSchedule = (filter) => async (dispatch) => {
+
+    await wait(2000);
```

Comment:
What's the wait for? also do you want to set loading to true before making the req?

Reply:
Good catch, I will change it so loading is set to false before the req.